File: backend/src/main/java/ru/mindflow/backend/entity/Meditation.java
package ru.mindflow.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "meditations")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Meditation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "duration_minutes")
    private Integer durationMinutes;

    @Column(name = "audio_url")
    private String audioUrl;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "difficulty_level")
    private String difficultyLevel;

    @Builder.Default
    @Column(nullable = false)
    private Boolean active = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @OneToMany(mappedBy = "meditation", cascade = CascadeType.ALL)
    private List<MeditationSession> sessions;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: android-app/app/src/main/java/ru/mindflow/app/entity/Meditation.kt
package ru.mindflow.app.entity

data class Meditation(
    val id: Long,
    val title: String,
    val description: String?,
    val durationMinutes: Int?,
    val audioUrl: String?,
    val imageUrl: String?,
    val difficultyLevel: String?,
    val categoryName: String?
) {
    fun formattedDuration(): String =
        durationMinutes?.let { "$it мин" } ?: "—"

    fun difficultyLabel(): String = when (difficultyLevel?.uppercase()) {
        "BEGINNER"     -> "Начинающий"
        "INTERMEDIATE" -> "Средний"
        "ADVANCED"     -> "Продвинутый"
        else           -> difficultyLevel ?: "—"
    }
}
